pyiceberg/table/inspect.py

```diff
@@ -157,74 +158,92 @@ def _readable_metrics_struct(bound_type: PrimitiveType) -> pa.StructType:
                 pa.field("readable_metrics", pa.struct(readable_metrics_struct), nullable=True),
             ]
         )
+        return entries_schema
 
+    def _get_entries(self, manifest: ManifestFile, discard_deleted: bool = True) -> "pa.Table":
+        import pyarrow as pa
+
+        schema = self.tbl.metadata.schema()
```

Comment:
something im concerned about with the `all_*` metadata tables is taking into account the table schema/partition evolution. 

since we're looking across all snapshots, it might not be right to use the current snapshot here